### PR TITLE
881 court date clear

### LINKED ISSUE #881: Court Date/Report info "clears" once date has passed
Part of epic #880 // dependent on #808 & #805

**What type of user is this for? [ volunteer/supervisor/admin/all or All CASA Admin]**
all

**Where does/should this occur?**
On the Edit CASA Case page (example: https://casa-qa.herokuapp.com/casa_cases/2/edit)
Screen Shot 2020-09-26 at 4 30 47 PM (2)

**Description**
After the Next Court Date has passed, that date should "clear" and display under "`past_court_dates` (see #805), and all other associated data should reset, including:
- `court_report_due_date`
- `court_report_submission` status

**Frontend**
After the Next Court date has passed, *no* date should appear next to text: Next Court Date on the `casa_case` details page, and all associated information above should also reset.

### PR DESCRIPTION
**Pull request by big-meel**
### What github issue is this PR for, if any?
Resolves #881 

### What changed, and why?
Created ".clear_court_dates" method in model
Added scope to filter out cases with passed dates
Added tests to "casa_case_spec.rb" 
Added task to scheduler.rake to update dates 

### How is this tested? (please write tests!) 💖💪
Tested with rspec

### CODE REVIEW

File: app/models/casa_case.rb [hunk 1]
```diff
@@ -35,6 +35,10 @@ class CasaCase < ApplicationRecord
       .where("birth_month_year_youth <= ?", 14.years.ago)
   }
 
+  scope :due_date_passed, -> {
```
Comment (compwron) [inline app/models/casa_case.rb:38]: yay scope :)

File: app/models/casa_case.rb [hunk 2]
```diff
@@ -65,6 +69,15 @@ def update_cleaning_contact_types(args)
     end
   end
 
+  def clear_court_dates
+    
+  update(court_date: nil,
```
Comment (compwron) [inline app/models/casa_case.rb:74]: This code should be more indented, but the linter will do it the next time we run it (see the readme for lint info). :)

File: app/models/casa_case.rb [hunk 3]
```diff
@@ -65,6 +69,15 @@ def update_cleaning_contact_types(args)
     end
   end
 
+  def clear_court_dates
```
Comment (compwron) [inline app/models/casa_case.rb:72]: I think the scope should be used in here, or else all court dates might get wiped... not just the ones in scope?

File: lib/tasks/scheduler.rake [hunk 4]
```diff
@@ -4,3 +4,10 @@ task transition_youth: :environment do
   CasaCase.should_transition.update_all(transition_aged_youth: true)
   puts "done."
 end
+
+desc "Clear court dates and report information when date has passed, run by heroku scheduler"
+task clear_passed_dates: :environment do
+  puts "Checking case due dates..."
+  CasaCase.due_date_passed.each { |key| key.clear_court_dates }
```
Comment (compwron) [inline lib/tasks/scheduler.rake:11]: :)

**Review by compwron — APPROVED** [review]
This works! Ina future PR, I want to make this safer by including a conditional in `.clear_court dates` so that someone doesn't call it by accident on a casa case which should not have its court date cleared! :)

File: app/models/casa_case.rb (hunk 2 above)
Comment (big-meel) [reply app/models/casa_case.rb:74]: It did look like something was a bit off. I had indented it earlier but at some point when making changes I may have missed that. *collapses in disgrace* Thanks for the feedback :)

**Comment by big-meel** [comment]
Thanks so much for this feedback :). going to look for another issue to tackle

**Pull request merged**